app/services/invoices/compute_taxes_and_totals_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Invoices
+  class ComputeTaxesAndTotalsService < BaseService
+    def initialize(invoice:, finalizing: true)
+      @invoice = invoice
+      @finalizing = finalizing
+
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: 'invoice') unless invoice
+
+      if customer_provider_taxation?
+        invoice.status = 'pending' if finalizing
+        invoice.tax_status = 'pending'
+        invoice.save!
+        after_commit { Invoices::ProviderTaxes::PullTaxesAndApplyJob.perform_later(invoice:) }
+
+        return result.unknown_tax_failure!(code: 'tax_error', message: 'unknown taxes')
```

Comment:
but then this is the expected behavior not a failure, isn't it?

Reply:
Invoice is not valid without taxes so that's why I wanted to mark this case with this failure. Probably it is just style preference. Other option could be that I check wherever I need if `invoice.tax_status = 'pending'` or something similar:)

Reply:
I don't want to block the change as you mentioned this is urgent and should be released before the next billing run... you can ignore my comments and go ahead.

However, I would avoid calling a failure some that is the normal behaviour of the service... this is for me a success with pending tax status as you mentioned. Failure should be used to indicate real failures like validation failures, etc.